### PR TITLE
SNOW-656580 support geometry data type in snowpark

### PR DESCRIPTION
**Pull request by sfc-gh-aalam**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes #SNOW-656580

2. Fill out the following pre-review checklist:

   - [x] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Geometry support was added to python connector in release 3.0.4. This PR update snowpark to use leverage the connector changes and add support for Geometry type.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,15 @@
 # Release History
 
+## 1.6.0 (TBD)
```
Comment (sfc-gh-stan) [inline CHANGELOG.md:3]: Question: why not include this in `1.5.0`?

**Comment by sfc-gh-stan** [comment]
Looks like `test_write_pandas_with_timestamps` is flaky.

File: CHANGELOG.md (hunk 1 above)
Comment (sfc-gh-aalam) [reply CHANGELOG.md:3]: Just too busy to get it out on time. I can target `1.5.1`

File: tox.ini [hunk 2]
```diff
@@ -11,6 +11,8 @@ source = src/snowflake/snowpark
          */src/snowflake/snowpark
          *\src\snowflake\snowpark
          */fips_env/lib/python*/site-packages/snowflake/snowpark
+[coverage:html]
+show_contexts = true
```
Comment (sfc-gh-stan) [inline tox.ini:15]: Nice! 🥳

**Review by sfc-gh-stan — APPROVED** [review]

**Review by sfc-gh-sfan — APPROVED** [review]

**Pull request merged**